src/gretel_synthetics/generate.py

```diff
@@ -175,18 +198,17 @@ def _predict_chars(model: tf.keras.Sequential,
         # using a categorical distribution to
         # predict the word returned by the model
         predictions = predictions / store.gen_temp
-        predicted_id = tf.random.categorical(
-            predictions, num_samples=1)[-1, 0].numpy()
+        predicted_id = tf.random.categorical(predictions, num_samples=1)[-1, 0].numpy()
 
         # We pass the predicted word as the next input to the model
         # along with the previous hidden state
         input_eval = tf.expand_dims([predicted_id], 0)
         sentence_ids.append(int(predicted_id))
 
         decoded = sp.DecodeIds(sentence_ids)
-        decoded = decoded.replace('<c>', ',')
+        decoded = decoded.replace("<c>", ",")
```

Comment:
@johntmyers we will want whatever delimiter exists in the input data to be encoded as a special token here. Two options would be: a) creating a dictionary mapping of common delimiters and tags that are passed to the sentencepiecetokenizer, and then reversed later. Or b) make the delimiter a BaseConfig item and creating a user defined token from that. See: https://github.com/gretelai/gretel-synthetics/issues/14

Reply:
The delimeter is a config option, see the `field_delimeter` attr that is defined on the base dataclass. Why do we need a different token? Assuming there's only one delimeter provided, can we just use the same token regardless of the delimeter char provided?

Reply:
I'm also assuming we need to update the annotation function that does this:

```python
  with open(store.input_data_path, 'r', encoding='utf-8', errors='replace') as infile:
        for line in infile:
            if store.max_lines and len(training_text) >= store.max_lines:
                break
            line = line.strip().replace(",", "<c>")
            training_text.append(line)
```

Reply:
My bad- I did not see the `field_delimiter` option in the base dataclass. 

As described above we'll need to replace the replace the delimiter once as a special character on annotation, specify the corresponding user defined token to the tokenizer, and then reverse the token with the delimiter in the text generation functions